exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -794,13 +793,16 @@ class BlobColumnType : ColumnType() {
     }
 
     override fun readObject(rs: ResultSet, index: Int) = when {
-        currentDialect is SQLServerDialect -> rs.getBytes(index)?.let(::ExposedBlob)
-        else -> rs.getBinaryStream(index)?.let(::ExposedBlob)
+        currentDialect is SQLServerDialect || currentDialect is SQLiteDialect ->
+            rs.getBytes(index)?.let(::ExposedBlob)
+
+        else -> rs.getBlob(index)?.binaryStream?.let(::ExposedBlob)
     }
 
     override fun setParameter(stmt: PreparedStatementApi, index: Int, value: Any?) {
         when (val toSetValue = (value as? ExposedBlob)?.inputStream ?: value) {
             is InputStream -> stmt.setInputStream(index, toSetValue)
+            is String -> stmt.setInputStream(index, toSetValue.byteInputStream())
```

Comment:
Changes to `setInputStream()` will require adding the property as an argument:
```kt
is InputStream -> stmt.setInputStream(index, toSetValue, useObjectIdentifier)
```

Could you detail a case when a String argument would be passed to `fillParameters()` for this column type, so this new branch would be reached?

Reply:
The `is String` branch was a remaining artifact from a previous solution attempt.